test/e2e/multicast_test.go

```diff
@@ -158,6 +170,553 @@ type multicastTestcase struct {
+		// sender1 sends 225.1.2.10 will be blocked by anp2 ruleName block-multicast-traffic. sender2 sends 225.1.2.10 will be not counted as well because igmp packets
+		// which allow receiver1 joins 225.1.2.10 was also blocked by anp1 rule name drop-igmp-report. multicast traffic which group 225.1.2.11 will be received by receiver1.
+		match, _ := regexp.MatchString(fmt.Sprintf("%s[[:space:]]+%s[[:space:]]+%d[[:space:]]+%d", data.testNamespace, receiverName1, sendSessions, 0), strings.TrimSpace(stdout))
+		if !match {
+			return false, nil
+		}
+		stdout, stderr, err = runAntctl(antreaPodReceiver2, antctlCmds, data)
+		t.Logf("Got %s multicast stats output: %s", antreaPodReceiver2, stdout)
+		if err != nil {
+			t.Errorf("Error when executing antctl get podmulticaststats, stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			return false, err
+		}
+		t.Logf("The result of running antctl get podmulticaststats in antreaPodReceiver2 is stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+		// some as receiver1 except igmp packets with group 225.1.2.10 will not be blocked.
+		match, _ = regexp.MatchString(fmt.Sprintf("%s[[:space:]]+%s[[:space:]]+%d[[:space:]]+%d", data.testNamespace, receiverName2, sendSessions*2, 0), strings.TrimSpace(stdout))
+		if !match {
+			return false, nil
+		}
+		stdout, stderr, err = runAntctl(antreaPodSender1, antctlCmds, data)
+		if err != nil {
+			t.Errorf("Error when executing antctl get podmulticaststats, stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			return false, err
+		}
+		t.Logf("The result of running antctl get podmulticaststats in antreaPodSender1 is stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+		match, _ = regexp.MatchString(fmt.Sprintf("%s[[:space:]]+%s[[:space:]]+%d[[:space:]]+%d", data.testNamespace, senderName1, 0, 0), strings.TrimSpace(stdout))
+		if !match {
+			return false, nil
+		}
+		stdout, stderr, err = runAntctl(antreaPodSender2, antctlCmds, data)
+		t.Logf("Got %s multicast stats output: %s", antreaPodSender2, stdout)
+		if err != nil {
+			t.Errorf("Error when executing antctl get podmulticaststats, stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+			return false, err
+		}
+		t.Logf("The result of running antctl get podmulticaststats in antreaPodSender2 is stdout: %s, stderr: %s, err: %v", stdout, stderr, err)
+		match, _ = regexp.MatchString(fmt.Sprintf("%s[[:space:]]+%s[[:space:]]+%d[[:space:]]+%d", data.testNamespace, senderName2, 0, sendSessions*2), strings.TrimSpace(stdout))
+		if !match {
+			return false, nil
+		}
+		stats, err := data.crdClient.StatsV1alpha1().AntreaNetworkPolicyStats(data.testNamespace).Get(context.TODO(), "anp1", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		t.Logf("Got AntreaNetworkPolicy stats: %v", stats)
+		if len(stats.RuleTrafficStats) != 3 {
+			return false, nil
+		}
+		if stats.RuleTrafficStats[0].TrafficStats.Packets == 0 {
```

Comment:
Since the rule number must be 3, why not use a for-loop to check the packets in each rule is not 0?

Reply:
updated